lib/matplotlib/backends/backend_cairo.py

```diff
@@ -171,14 +171,18 @@ def draw_image(self, gc, x, y, im):
         # bbox - not currently used
         if _debug: print('%s.%s()' % (self.__class__.__name__, _fn_name()))
 
-        rows, cols, buf = im.color_conv (BYTE_FORMAT)
-        surface = cairo.ImageSurface.create_for_data (
-                      buf, cairo.FORMAT_ARGB32, cols, rows, cols*4)
+        if sys.byteorder == 'little':
+            im = im[:, :, (2, 1, 0, 3)]
+        else:
+            im = im[:, :, (3, 0, 1, 2)]
```

Comment:
RGBA is always passed to draw_image (it's the responsibility of the other layers, not the backend, to make it so).  This should be documented as well in `BackendBases.draw_image`.

I checked the swaps on both x86_64 and an old PPC Mac-running-Linux that I keep around to check byte order issues.  Cairo is the only backend we have that requires ARGB in native byte order like this.